test/performance/scale_from_zero_test.go

```diff
@@ -57,21 +59,24 @@ func runScaleFromZero(idx int, t *testing.T, clients *test.Clients, ro *test.Res
 		test.DeploymentScaledToZeroFunc,
 		"DeploymentScaledToZero",
 		test.ServingNamespace,
-		2*time.Minute); err != nil {
+		3*time.Minute); err != nil {
 		m := fmt.Sprintf("%02d: failed waiting for deployment to scale to zero: %v", idx, err)
 		t.Log(m)
 		return 0, errors.New(m)
 	}
 
 	start := time.Now()
-	t.Log("Waiting for endpoint to serve request")
-	if _, err := pkgTest.WaitForEndpointState(
+	t.Logf("%0d: waiting for endpoint to serve request", idx)
+	if _, err := pkgTest.WaitForEndpointStateWithTimeout(
 		clients.KubeClient,
 		t.Logf,
 		domain,
-		test.RetryingRouteInconsistency(pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody(helloWorldExpectedOutput))),
+		pkgTest.Retrying(pkgTest.MatchesAllOf(pkgTest.IsStatusOK, pkgTest.MatchesBody(helloWorldExpectedOutput)),
```

Comment:
I'm still unclear on why we're willing to retry 404 and 500 vs. simply expecting things to work.

cc @markusthoemmes @tcnghia in case they have any ideas.

In general, I think I'm fine with the cleanup here, but I'd like to see us open an issue to track finding and fixing the reason we need retries here.

Reply:
Well, in fact it takes quite some time for errors to go away. Up to 2 minutes in some cases.

Reply:
Hm, the retry that you added now should not be a retry at all? You're not retrying any status codes and you force a 200 + a specific body. Dropping the retry should be fine in this case.

I agree with @mattmoor we need to root cause any unexpected 503s, but we also need to be very clear about when they happen and what their body is. The root causes will potentially be very very different.

Reply:
Nevermind the comment above.

I'm also unclear why we see 404s in this case. If we do, that's **really** unexpected. The 500s is something we see in our e2e tests generally. I have the feeling that our default probe is not as effective as we think it is.

In any case, as this already had a "WaitForEndpointState": Would you add a TODO and open an issue pointing at this TODO to root cause and debug please? Please be specific about the errors seen (no healthy upstream vs. close before headers makes quite a difference here). We don't necessarily need to block this specific PR any longer on this if it's a genuine bug and you're seeing these errors.

Reply:
As such, the changes we're proposing for handoff, should remove the requirements for this retries.
For now I'm going to remove the retries altogether and Skip the 50 test back.